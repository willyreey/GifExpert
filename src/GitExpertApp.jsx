import {useState} from 'react'
import { AddCategory,GifGrid } from './components/'


export const GitExpertApp = () => {

    const [categories, setCategories] = useState([])
    
    const onAddCategory = (newCategory) =>{
      if( categories.includes(newCategory) ) return
      setCategories([newCategory,...categories])
    }

    

    return (
      <>

      {/* Titulo */}
      <h1>GitExpertApp</h1>

      {/* Input */}
      <AddCategory 
      onNewCategory={onAddCategory}

      />

      {/* Listado de Gif */}
      {
        categories.map( (category) => (

          <GifGrid 
            key={category}
            category={category}
          />
        ))
      }

      </>
    )
  }