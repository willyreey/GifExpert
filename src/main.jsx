import React from 'react'
import ReactDOM from 'react-dom/client'
import { GitExpertApp } from './GitExpertApp.jsx'
import './styles.css'


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GitExpertApp />
  </React.StrictMode>,
)
