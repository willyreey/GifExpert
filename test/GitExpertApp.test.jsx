import { render } from "@testing-library/react"
import { GitExpertApp } from "../src/GitExpertApp"

describe('Probando GitExpertApp', () => { 
    
    test('Debe de hacer match con el snapshot', () => { 
        
        const { container } = render(<GitExpertApp/>)
        expect(container).toMatchSnapshot()

     })
    
 })