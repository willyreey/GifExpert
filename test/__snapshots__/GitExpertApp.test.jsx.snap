// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GitExpertApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    GitExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gif"
      type="text"
      value=""
    />
  </form>
</div>
`;
